submissions/ik-web/oop-exercise/index.js

```diff
@@ -0,0 +1,65 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/ik-web/a-tiny-JS-world
+   Web app: https://ik-web.github.io/a-tiny-JS-world/
+
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+   }
+
+   showInhabitantInfo() {
+      return `A ${this.species} (<i>${this.gender}<i>) named <strong>${this.name}</strong>, has ${this.legs} legs, likes to say <u>${this.saying}</u>.`;
+   }
+}
+
+class Dog extends Inhabitant {
+   constructor(name, gender) {
+      super('dog', name, gender, 4, 'woof-woof');
+   }
+}
+
+class Cat extends Inhabitant {
+   constructor(name, gender) {
+      super('cat', name, gender, 4, 'meow');
+   }
+
+   getSaying() {
+      return this.saying;
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, hands, saying) {
+      super('human', name, gender, 2, saying);
+      this.hands = hands;
+   }
+
+   showInhabitantInfo() {
+      return super.showInhabitantInfo().replace('legs,', `legs and ${this.hands} hands,`);
+   }
+}
+
+class CatWoman extends Human {
+   constructor(name, saying) {
+      super(name, 'female', 2, saying);
+      this.species = 'mystery of nature';
+   }
+}
+
+const dog = new Dog('Bobik', 'male');
+const cat = new Cat('Barsik', 'male');
+const man = new Human('Benjamin', 'male', 2, 'Hi to all!');
+const woman = new Human('Sarah', 'female', 2, 'Hello everyone!');
+const catWoman = new CatWoman('anonymos', cat.getSaying());
```

Comment:
Here a cat-woman borrows saying from a particular cat. What if we've got more than 1 cat? What if we've got no cats at all?
If it was supposed to borrow saying from any cat default saying then [class static properties and methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) are to the rescue.

Reply:
Thank you for the help and advice